mesheryctl/internal/cli/root/perf/perf.go

```
@@ -129,79 +127,75 @@ var PerfCmd = &cobra.Command{
 			}
-			if qps == "" {
-				qps = fmt.Sprintf("%f", t.Client.Rps)
+		} else {
+
+			if testName == "" {
+				log.Debug("Test Name not provided")
+				testName = StringWithCharset(8)
+				log.Debug("Using random test name: ", testName)
 			}
-		}
 
-		if testName == "" {
-			log.Debug("Test Name not provided")
-			testName = StringWithCharset(8)
-			log.Debug("Using random test name: ", testName)
-		}
+			postData := ""
 
-		postData := ""
+			startTime := time.Now()
+			duration, err := time.ParseDuration(testDuration)
+			if err != nil {
+				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to parse test duration %s", testDuration)))
+			}
 
-		startTime := time.Now()
-		duration, err := time.ParseDuration(testDuration)
-		if err != nil {
-			return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to parse test duration %s", testDuration)))
-		}
+			endTime := startTime.Add(duration)
 
-		endTime := startTime.Add(duration)
+			postData = postData + "start_time: " + startTime.Format(time.RFC3339)
+			postData = postData + "\nend_time: " + endTime.Format(time.RFC3339)
 
-		postData = postData + "start_time: " + startTime.Format(time.RFC3339)
-		postData = postData + "\nend_time: " + endTime.Format(time.RFC3339)
+			if testURL != "" {
+				postData = postData + "\nendpoint_url: " + testURL
+			} else {
+				return errors.New(utils.PerfError("please enter a test URL"))
+			}
 
-		if testURL != "" {
-			postData = postData + "\nendpoint_url: " + testURL
-		} else {
-			return errors.New(utils.PerfError("please enter a test URL"))
-		}
+			// Method to check if the entered Test URL is valid or not
+			var validURL = govalidator.IsURL(testURL)
 
-		// Method to check if the entered Test URL is valid or not
-		var validURL = govalidator.IsURL(testURL)
+			if !validURL {
+				return errors.New(utils.PerfError("please enter a valid test URL"))
```

@kanishkarj Can we also log the errors using logrus ?

---

I think error logging is handled separately... I am not sure who, But someone who restructured the commands to return errors instead of logging them?

---

@nitishm maybe?

---

> @nitishm maybe?

Yup I switched to cobra's RunE function.

---

I don't exactly know how cobra's RunE fn works, @nitishm If I log it separately will it get redundant?